UI Changes & Critical Bug Fix & Backend Refactor & Version Manager

OmniacDev opened the pull request:
Fixed a bug where the proxy wasn't able to be located due to recent changes in the bundler

and there's a bunch of UI changes cause I accidentally did them on the same branch :P

OmniacDev committed:
UI updates

OmniacDev committed:
Merge pull request #7 from OmniacDev/updates

UI updates

OmniacDev committed:
Merge pull request #6 from FrederoxDev/main

Merge from main

OmniacDev committed:
ui updates

OmniacDev committed:
UI updates

OmniacDev committed:
Fixes

OmniacDev committed:
+

OmniacDev committed:
don't need that

OmniacDev committed:
moved stuff around

OmniacDev committed:
CSS

OmniacDev committed:
stuff idk

OmniacDev committed:
.

FrederoxDev commented on src/renderer/versionSwitcher/AppRegistry.ts:
```diff
@@ -57,10 +57,11 @@ export async function unregisterExisting() {
 }
 
 export async function registerVersion(version: MinecraftVersion) {
+    const maxAttempts = 100;
```
this can be set to like 30 or something

FrederoxDev replied:
If it fails, it would be helpful to display the fix to this in launcher. The solution is pinned in the support channel in the discord rn. At no point though should the launcher automatically do any of it, I want to keep any uninstall steps completely on the end users doing

FrederoxDev reviewed (changes requested):
see comment about max attempts. Can I also get screenshots of any UI changes

OmniacDev committed:
maxAttempts

OmniacDev committed:
UI stuff

OmniacDev commented:
### **Global UI Changes:**

- Navbar and View containers are now floating
- Background from home page is now applied to the main view container and sits behind all floating containers
- Everything now has 3px borders for consistency

### **Titlebar Changes**: 
- Added a 3px shadow

### **Home Page Changes:**
- Moved background to the main view container

![Screenshot 2024-07-08 231641](https://github.com/FrederoxDev/Amethyst-Launcher/assets/64994204/38b0c9e8-a1f0-4f79-bafa-454edfd8c795)

### **Profiles Page Changes:**
- Improved profile buttons

![Screenshot 2024-07-08 231650](https://github.com/FrederoxDev/Amethyst-Launcher/assets/64994204/baf0a6c4-8244-4870-81ae-3303d1af3ddc)

### **Mods Page Changes:**
- Edited buttons to match those in the Profiles Page

![Screenshot 2024-07-08 231657](https://github.com/FrederoxDev/Amethyst-Launcher/assets/64994204/2e74915d-ad1a-45f0-bdbe-98d12bc89cab)

### **Settings Page Changes:**
- Made settings container scrollable
- Changed the settings section to the correct style

**Other changes:**
- Refactored some of the components, added themes for dark and light mode to the MinecraftRadialButton component.

![Screenshot 2024-07-08 231704](https://github.com/FrederoxDev/Amethyst-Launcher/assets/64994204/d1dd1efc-706b-4092-87d7-e1d1e5560222)
![Screenshot 2024-07-08 231713](https://github.com/FrederoxDev/Amethyst-Launcher/assets/64994204/cbfa48cb-15f4-4663-9a8a-116fc391bd74)

OmniacDev committed:
these random spaces bruh

OmniacDev committed:
oopsie

OmniacDev committed:
another lil oopsie

OmniacDev committed:
bruh

OmniacDev committed:
hmm

FrederoxDev commented:
I noticed that the font for the launcher config debug viewer seems to be wrong

OmniacDev commented:
yeah I changed it to be a bit more readable since it could be important

OmniacDev commented:
want it changed back?

FrederoxDev commented:
Yes please, I dont really find it unreadable at its current state?

OmniacDev committed:
Fix "Launch Game" Hang

OmniacDev committed:
Revert font change on text-box

OmniacDev commented:
Done 👍🏻

OmniacDev commented:
![image](https://github.com/FrederoxDev/Amethyst-Launcher/assets/64994204/9350b0fa-6ee5-4347-8c91-f9991774d8a9)

OmniacDev committed:
Refactoring ModList.ts and ModConfig.ts into Mods.ts

OmniacDev committed:
Pruning & Moving

OmniacDev committed:
Refactored Paths.ts

OmniacDev committed:
Disable CORS

OmniacDev commented:
https://www.youtube.com/watch?v=5xooMXyleXM

OmniacDev commented:
oh shit didn't mean to close it

OmniacDev committed:
Fix Warnings

OmniacDev committed:
Disable Electron's Security Warnings

OmniacDev committed:
Refactor a few files

OmniacDev committed:
Refactor a few files

OmniacDev committed:
Finished Refactoring Backend

OmniacDev committed:
Consistent Button Styles

OmniacDev committed:
Consistent Border Styles

OmniacDev committed:
Merge pull request #8 from OmniacDev/refactor

Merge Refactor Branch

OmniacDev commented:
Refactored a lot of the launcher's backend and fixed a few issues.

OmniacDev committed:
Fix AppData Paths

OmniacDev committed:
Use vite for building electron's files, added version manager page to display currently installed versions.

OmniacDev committed:
cleanup package

OmniacDev commented:
PR is getting a bit big

OmniacDev committed:
Versions stuff

OmniacDev committed:
Implement versions.json file read & write

OmniacDev committed:
Basic VersionManager working

OmniacDev commented:
### Version Manager

Added Version Manager page which automatically finds versions within Amethyst's AppData folder, and displays them.
Currently there are three buttons;
- Delete version
- Open version folder
- Version info (pretty basic right now, just shows the version folder path)


https://github.com/FrederoxDev/Amethyst-Launcher/assets/64994204/d36dba56-d3ec-4e00-ad0f-7f3fdc6d96d6

OmniacDev committed:
Improved Error Popup UI, changed unregister packages command to use PreserveRoamableApplicationData isntead of PreserveApplicationData so unregistering doesn't fail when a version from MS Store is installed.

OmniacDev commented:
# Error Popup UI:
- Improved and added close button

![image](https://github.com/user-attachments/assets/c3fa52b0-7c4d-49ff-9393-b5ae9888e2a0)


# AppRegistry Changes:
- **Unregister Package:** Replaced `PreserveApplicationData` arg with `PreserveRoamableApplicationData`, so that the unregistering doesn't fail when a version from MS Store is installed.

**Note:** All necessary AppData seems to be preserved, but more testing may be required to confirm this.

OmniacDev committed:
Remove MoveVersion from version button

OmniacDev committed:
Created PopupPanel.tsx

FrederoxDev commented:
Ok, well im willing to merge this whenever you're ready, it seems there are some conflicts from another PR

FrederoxDev commented:
Also did you remove Ctrl R? just noticed its not working in the launcher anymore.

OmniacDev committed:
Import .zip and folders

OmniacDev committed:
Changed assumed .zip structure, error popup improvements

OmniacDev commented:
# Import .ZIP Files and Folders
- Modified and added functionality to drag and drop .zip files and folders into the launcher from PR #45 
- Fixed a lot of stuff in it.

# Error Popup UI
- Made it more generic, and added a warning icon


https://github.com/user-attachments/assets/19a4d559-3f6f-45cc-8d97-7860c9ad2e85

OmniacDev committed:
Merge branch 'main' of https://github.com/FrederoxDev/Amethyst-Launcher

OmniacDev committed:
Revert "Merge branch 'main' of https://github.com/FrederoxDev/Amethyst-Launcher"

This reverts commit 94a03ac559fc346e89c5e1a8dc8d56eb1cae002f, reversing
changes made to 965b34aefd53a9e4aa8b5f443289bf9a24af5b92.

OmniacDev committed:
Added reload back

OmniacDev committed:
Improved button styles

OmniacDev commented:
Imma start working on some new stuff on a separate branch so this PR is pretty much done, lmk if it needs any other changes

FrederoxDev commented:
i'll test and merge this later on today
